cwm_worker_tests/upload_download_test.py

```
@@ -42,44 +39,61 @@ def upload(endpoint, access_key, secret_key, bucket, num_files, file_size, outpu
         else:
             print(f'Bucket exists! [{bucket}]')
 
-        start_time = datetime.datetime.now()
-
-        # Create test file, remove at the end
-        filepath = create_test_file(file_size)
-
-        print(f'Uploading files...')
-        for i in range(num_files):
-            filename = get_upload_filename()
-            client.fput_object(bucket, filename, filepath)
-            print(f'\rUploaded: {i+1} files', end='', flush=True)
-
-        end_time = datetime.datetime.now()
-        elapsed_seconds = (end_time - start_time).total_seconds()
-        print(f'\nFiles uploaded! [{elapsed_seconds} seconds]')
-
+        test_filepath = create_test_file(file_size)
+
+        timestamp = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.000%fZ')
+        output_filename = f'upload-report-{timestamp}.csv'
+        output_filepath = output_dir + '/' + output_filename
+
+        with open(output_filepath, 'w', buffering=1) as csvfile:
+            csvfile.write(f'file_index,filename,file_upload_elapsed_time_seconds\n')
+            print(f'Uploading files... [CSV report: {output_filepath}]')
+            for i in range(num_files):
+                index = i+1
+                file_upload_start_time = datetime.datetime.now()
+                filename = get_filename(file_size=file_size, index=index)
+                try:
+                    client.fput_object(bucket, filename, test_filepath)
+                except S3Error as e:
+                    print(f'Exception: {e}')
+                    exit(1)
+                except Exception as e:
+                    print(f'Exception: {e}')
+                    exit(1)
```

you should remove this try catch and just let the errors be raised